code/game/machinery/doors/firedoor.dm

```diff
@@ -83,7 +78,7 @@
 /obj/machinery/door/firedoor/LateInitialize()
 	. = ..()
 	GetMergeGroup(merger_id, allowed_types = merger_typecache)
-
+	RegisterSignal(GetMergeGroup(merger_id, allowed_types = merger_typecache), COMSIG_MERGER_REFRESH_COMPLETE, .proc/refresh_shared_turfs)
```

Comment:
GetMergeGroup returns the var/datum/merger that it creates, stick that in a variable, and just feed it into RegisterSignal instead of calling it twice